client/client.go

```diff
@@ -349,11 +349,12 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
+		resp.Body.Close()
 		<-done
-		if err == nil {
-			err = ctx.Err()
-		}
+		// Because the body is closed, the err var from ReadAll goroutine is
```

Comment:
I would expect not nil error + empty response/body. Or nil err + non-empty response/body. For the race, we do not have to care about who wins. We just want a consistent result. Right?